src/core/border_router/routing_manager.cpp

```diff
@@ -326,6 +390,37 @@ void RoutingManager::HandleReceived(const InfraIf::Icmp6Packet &aPacket, const I
     return;
 }
 
+Error RoutingManager::HandleReceivedHostAddress(const char *        aHostName,
+                                                const Ip6::Address *aAddresses,
+                                                uint8_t             aNumAddresses)
+{
+    Error       error = kErrorNotImplemented;
+    Ip6::Prefix prefix;
+
+    OT_UNUSED_VARIABLE(aHostName);
+    OT_UNUSED_VARIABLE(aAddresses);
+    OT_UNUSED_VARIABLE(aNumAddresses);
+    OT_UNUSED_VARIABLE(prefix);
+
+    VerifyOrExit(aHostName == kWellKnownIpv4OnlyName, error = kErrorNotImplemented);
```

Comment:
```suggestion
    VerifyOrExit(aHostName == kWellKnownIpv4OnlyName);
```
error is `kErrorNotImplemented` by default